java/pom.xml

```diff
@@ -231,6 +231,45 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>COPY_NVRTC</id>
+            <activation>
+                <property>
+                    <name>CUDA_STATIC_RUNTIME</name>
+                    <value>ON</value>
+                </property>
+            </activation>
+            <build>
+            <plugins>
+              <plugin>
+                <artifactId>maven-resources-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>copy-nvrtc</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>copy-resources</goal>
+                        </goals>
+                        <configuration>
+                            <overwrite>true</overwrite>
+                            <skip>${skipNativeCopy}</skip>
+                            <outputDirectory>${basedir}/target/native-deps/${os.arch}/${os.name}</outputDirectory>
+                            <resources>
+                                <resource>
+                                    <!--Set by groovy script-->
+                                    <directory>${cuda.path}</directory>
+                                    <includes>
+                                        <include>libnvrtc.so</include>
```

Comment:
I don't see the benefit of including libnvrtc.so without the builtin.  We know it needs that library at runtime, so why include only half here?  If we're expecting the builtins lib to be on the machine then it's safe to assume `libnvrtc` is there as well.

Also are we sure `libnvrtc` or `libnvrtc-builtins` do not require a specific CUDA runtime when they load?  If they do, then we either need to bundle the runtime or it's not worth trying to bundle these.

Reply:
Then we should just rip out all of the static linking support. As I said this is a step in the right direction.  Including libnvrtc-builtins does not work. nvrtc will load it through dlopen and bypass the version that we already loaded. I could see this while looking at `/proc/maps` for the process.  No cudart ever is loaded, but nvrtc-builtins shows up a little while after I delete the cudf jit cache, even if we are packaging/loading -builtins.

Reply:
I'm trying to see the "step" here.  In what situation will including libnvrtc.so in the jar help?  It seems like we're just inflating the jar size here for no benefit.

Reply:
As it is right now there is no situation where turning on static linking helps.  Which is why I said we should just remove all of it, if you don't think this is a step in the right direction. The java library is still tied to a specific runtime being installed through libnvrtc. My thought was that we can start working with the nvrtc team to understand how we can work around loading builtins. The change does not break anything.

Reply:
It doesn't break anything, but it doesn't help either.  IMO we should add this code when it helps something, otherwise it's just dead code for zero benefit (or negative benefit if we bundle it).  If and when we can support bundling libnvrtc successfully, that's when we should add it to the jar.  It's not like this is a ton of code.

Reply:
I think we should focus this PR on removing the nvstrings dependency.  We can discuss what to do with libnvrtc and static linking in another PR.